include/swift-c/SyntaxParser/SwiftSyntaxParser.h

```diff
@@ -232,6 +232,65 @@ swiftparse_parse_string(swiftparse_parser_t, const char *source);
 /// declarations, etc.
 SWIFTPARSE_PUBLIC const char* swiftparse_syntax_structure_versioning_identifier(void);
 
+typedef struct {
+  /// Represents the range for the fixit.
+  swiftparse_range_t range;
+  /// Represent the text for replacement.
+  const char* text;
+} swiftparse_fixit_t;
+
+enum swiftparser_diagnostic_severity {
+  swiftparser_diagnostic_severity_Error = 0,
+  swiftparser_diagnostic_severity_Warning = 1,
+  swiftparser_diagnostic_severity_Remark = 2,
+  swiftparser_diagnostic_severity_Note = 3,
+};
+
+/// This is for the client to provide an opaque pointer that the parser will
+/// associate with a diagnostic.
+/// This pointer is only valid when received from the argument of a
+/// swiftparse_diagnostic_handler_t block and inside the block.
+typedef void* swiftparser_diagnostic_t;
+
+/// Invoked by the parser when a diagnostic is emitted.
+typedef void(^swiftparse_diagnostic_handler_t)(const swiftparser_diagnostic_t);
+
+/// Set the \c swiftparse_diagnostic_handler_t block to be used by the parser.
+///
+/// It isn't required to set a \c swiftparse_diagnostic_handler_t block.
+SWIFTPARSE_PUBLIC void
+swiftparse_parser_set_diagnostic_handler(swiftparse_parser_t,
+                                         swiftparse_diagnostic_handler_t);
+
+/// Get the message of a swiftparser_diagnostic_t
+SWIFTPARSE_PUBLIC const char*
+swiftparse_diagnostic_get_message(swiftparser_diagnostic_t);
+
+/// Get the source location in byte offset to where the diagnostic is issued
+/// in the source buffer.
+SWIFTPARSE_PUBLIC
+unsigned swiftparse_diagnostic_get_source_loc(swiftparser_diagnostic_t diag);
+
+/// Get the number of fixits of a swiftparser_diagnostic_t
+SWIFTPARSE_PUBLIC unsigned
+swiftparse_diagnostic_get_fixit_count(swiftparser_diagnostic_t);
+
+/// Get the fixit at the specified index of a swiftparser_diagnostic_t
+SWIFTPARSE_PUBLIC swiftparse_fixit_t
+swiftparse_diagnostic_get_fixit_at_index(swiftparser_diagnostic_t, unsigned);
+
+/// Get the number of highlight ranges of a swiftparser_diagnostic_t
+SWIFTPARSE_PUBLIC unsigned
+swiftparse_diagnostic_get_range_count(swiftparser_diagnostic_t);
+
+/// Get the highlight range at the specified index of a swiftparser_diagnostic_t
+SWIFTPARSE_PUBLIC swiftparse_range_t
+swiftparse_diagnostic_get_range_at_index(swiftparser_diagnostic_t, unsigned);
+
+/// Get the severity of a swiftparser_diagnostic_t
+SWIFTPARSE_PUBLIC enum swiftparser_diagnostic_severity
```

Comment:
If you write the enum as typedef as in `sourcekitd.h` you can remove the `enum` tag from the type here.